protocol/consistencychecks.go

```diff
@@ -278,30 +329,33 @@ func (cc *ConsistencyChecks) verifySignedPromise(requestType int,
 	case RegistrationType:
 		if responseCode == ErrorNameExisted {
 			// ignore the key value verification since we may have no information about that existed binding.
-			if verifResult := cc.verifyReturnedPromise(tb, uname, nil, str, ap); verifResult != nil {
+			if verifResult := cc.verifyTB(tb, uname, nil, str, ap); verifResult != nil {
```

Comment:
Alternatively, the client can check if the newly received TB for its user matches the existing cached TB for its user.